### PR TITLE
Fixes/#221 include decommission date for generators

### LINKED ISSUE #221: Missing Filter for EinheitBetriebsstatus in power_plants_status_quo Function
### Issue Description
In the [`power_plants_status_quo`](https://github.com/openego/powerd-data/blob/bad9f704bce388abaf0701401c17724d182717e4/src/egon/data/datasets/power_plants/__init__.py#L907-L1229) function, it appears that there's a missing filter for the `EinheitBetriebsstatus` attribute to ensure that only power plants with a status of "InBetrieb" (In Operation) are represented in the model. Currently, decommissioned conventional power plants are also included, which leads to inaccuracies in the energy model's outcomes.

### Expected Behavior
The function should include a filter that selects only the power plants with an `EinheitBetriebsstatus` of "InBetrieb" to ensure that the model accurately represents operational power plants only.

### Current Behavior
Decommissioned conventional power plants are represented in the model due to the absence of a filter for the `EinheitBetriebsstatus` attribute in the `power_plants_status_quo` function.

### Possible Solution
Implement a filter in the `power_plants_status_quo` function that checks the `EinheitBetriebsstatus` attribute of each power plant and includes only those with a status of "InBetrieb" in the model.

### PR DESCRIPTION
**Pull request by CarlosEpia**
Fixes #221.

### CODE REVIEW

File: src/egon/data/datasets/power_plants/__init__.py [hunk 1]
```diff
@@ -908,6 +909,8 @@ def power_plants_status_quo(scn_name="status2019"):
     con = db.engine()
     session = sessionmaker(bind=db.engine())()
     cfg = egon.data.config.datasets()["power_plants"]
+    year = int(scn_name[-4:])
```
Comment (khelfen) [inline src/egon/data/datasets/power_plants/__init__.py]: `scenario_date_max = f"{scn_name[-4:]}-12-31 23:59:00"`

File: src/egon/data/datasets/power_plants/__init__.py [hunk 2]
```diff
@@ -1014,48 +1044,43 @@ def fill_missing_bus_and_geom(gens, carrier):
...
+
+    # convert from KW to MW
+    conv["capacity"] = conv["capacity"] / 1000
+
+    # drop chp generators
+    conv["ThermischeNutzleistung"] = conv["ThermischeNutzleistung"].fillna(0)
+    conv = conv[conv.ThermischeNutzleistung == 0]
+
+    # rename carriers
+    conv.loc[conv.carrier == "Braunkohle", "carrier"] = "lignite"
```
Comment (khelfen) [inline src/egon/data/datasets/power_plants/__init__.py]: A dictionary plus [`pandas.Series.map`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.map.html) would be cleaner

**Review by khelfen — COMMENTED** [review]
I just made two small suggestions and the part for pv rooftop is missing. I did not test the code, but it looks like it should work :)

**Comment by CarlosEpia** [comment]
> I just made two small suggestions and the part for pv rooftop is missing. I did not test the code, but it looks like it should work :)

@khelfen What do you think if we create a new issue to deal with the new filters for generators in future scenarios?
I would like to close this issue related to the "status" scenarios soon and changing all the scrips for eGon2035 and eGon100RE looks like a task that could take some time

**Comment by khelfen** [comment]
> > I just made two small suggestions and the part for pv rooftop is missing. I did not test the code, but it looks like it should work :)
> 
> @khelfen What do you think if we create a new issue to deal with the new filters for generators in future scenarios? I would like to close this issue related to the "status" scenarios soon and changing all the scrips for eGon2035 and eGon100RE looks like a task that could take some time

That's fine with me :)

File: src/egon/data/datasets/power_plants/__init__.py (hunk 1 above)
Comment (CarlosEpia) [reply src/egon/data/datasets/power_plants/__init__.py]: Implemented in 341db612b04a6fcd966a876ce78dfebde3ed31da

File: src/egon/data/datasets/power_plants/__init__.py (hunk 2 above)
Comment (CarlosEpia) [reply src/egon/data/datasets/power_plants/__init__.py]: Implemented in 22b275de6172fb7f60c71459a2266a2be0e76b18

**Comment by CarlosEpia** [comment]
Since the comments were already addressed, I proceed to merge this branch.

**Pull request merged**